tests/test_pipeline_output.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+import pandas as pd
+import os
+import allensdk.core.json_utilities as ju
+from allensdk.ipfx.bin.run_pipeline import run_pipeline
+from pkg_resources import resource_filename
+
+if 'TEST_SPECIMENS_FILE' in os.environ:
+    TEST_SPECIMENS_FILE = os.environ['TEST_SPECIMENS_FILE']
+else:
+    TEST_SPECIMENS_FILE = resource_filename(__name__, 'test_specimens.csv')
+
+test_specimens = pd.read_csv(TEST_SPECIMENS_FILE, sep=" ")
+test_specimens_params = [tuple(sp) for sp in test_specimens.values]
+
+
+def check_json_files_are_equal(benchmark_json,test_json):
+
+    benchmark_pipeline_output = ju.read(benchmark_json)
+    test_pipeline_output = ju.read(test_json)
+
+    return benchmark_pipeline_output == test_pipeline_output
+
+
+@pytest.mark.parametrize('specimen,benchmark_pipeline_input_json,benchmark_pipeline_output_json', test_specimens_params)
+def test_pipeline_output(specimen, benchmark_pipeline_input_json, benchmark_pipeline_output_json, tmpdir_factory):
+    """
+    Runs pipeline, saves to a json file and compares to the existing pipeline output.
+    Raises assertion error if the dicts with pipeline outputs are not identical.
+
+    Parameters
+    ----------
+    specimen: string/int specimen name/id
+    benchmark_pipeline_input_json: string json file name
+    benchmark_pipeline_output_json: string json file name
+    tmpdir_factory: pytest fixture
+
+    Returns
+    -------
+
+    """
+
+    print (specimen,benchmark_pipeline_input_json,benchmark_pipeline_output_json)
+
+    bpi = ju.read(benchmark_pipeline_input_json)
+    test_dir = str(tmpdir_factory.mktemp("%s" % specimen))
+
+    tpi = dict(bpi)
```

Comment:
I think you're trying to make a copy here.  1) This will only do a shallow copy -- you probably want `copy.deepcopy`.  2) Do you need to make a copy?  `bpi` isn't used anywhere else.